esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -4011,8 +4051,11 @@ IFileIOStream* CWsWuFileHelper::createWUFileIOStream(IEspContext& context, const
     return createIOStreamWithFileName(zipFileNameWithPath.str(), IFOread);
 }
 
-void CWsWuFileHelper::validateFilePath(const char *file, bool UNCFileName, const char *fileType, const char *compType, const char *compName)
+void CWsWuFileHelper::validateFilePath(const char* file, WsWuInfo& winfo, CWUFileType wuFileType, bool UNCFileName, const char* fileType, const char* compType, const char* compName)
 {
+    if (validateWUFile(file, winfo, wuFileType))
```

Comment:
Is there ever a valid case where this should returns false and the user is accessing a file they should?  That is, the file they ask for should have been listed in the workunit, correct?

Just want clarification just in case.  I'm wondering if a return of false above should ever be allowed to continue below?  Doesn't that almost imply something is wrong and they may be trying to access files in a way that they shouldn't?  A file outside the scope of this workunit?

If there are some files that need to be accessed that aren't found in validateWUFile() then ok we just need to control where they can access.  but otherwise, I think the narrower the scope of access the better.

Reply:
@jakesmith @wangkx I think Jake and I had similar thoughts but different conclusions. His suggestion of validating the real path does solve the immediate problem you were having.

But I do think validating onWUFile requests (not other ways of getting files) to make sure you have access to the workunit and the file you are asking for is referenced in the workunit is a stronger form of security checking in the end. 

A user can't fish for files related to another workunit.. or other files.

Perhaps we can fix the realpath checking now and consider restricting further in the future.

Reply:
Yes, we can fix the realpath checking now and consider restricting further in the future. @jakesmith any comment? Could you please let me know how to convert the path (such as /var/log/...) to their real paths (such as /mnt/disk1/var/log/...)?

Reply:
> But I do think validating onWUFile requests (not other ways of getting files) to make sure you have access to the workunit and the file you are asking for is referenced in the workunit is a stronger form of security checking in the end.
> 
> A user can't fish for files related to another workunit.. or other files.

I agree, it makes sense to limit valid requests to files only within the workunit they are being request for.
So I think this change makes sense, even if there are other issues (like potential out of scope files via topology), that might need tackling differently, e.g. with help from realpath.